tests/console/validate_partitioning.pm

```diff
@@ -0,0 +1,89 @@
+# Suse'S openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Simple LVM partition validation
+# Maintainer: Yiannis Bonatakis <ybonatakis@suse.com>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use utils;
+use y2logsstep;
+
+sub run {
+    my $self     = shift;
+    my $expected_lv_stats = {
+        write_access => qr/\s{2}LV Write Access \s+ read\/write/,
+        status       => qr/\s{2}LV Status \s+ available/,
+        readahead    => qr/\s{2}Read ahead sectors \s+ auto/,
+	testactive       => qr/\s{2}# open \s+ [12]/,
+        # 254 as major no. points to dev-mapper, see /proc/devices
+        block_device => qr/\s{2}Block device \s+ 254:\d/
+    };
+
+    $self->select_serial_terminal;
+    
+    record_info('INFO', 'Validate LVM config');
+    assert_script_run "lvmconfig --mergedconfig --validate | grep \"LVM configuration valid.\"";
+        
+    record_info('INFO', 'Print lvm setup');
+    assert_script_run 'lsblk';
+    assert_script_run 'lvmdiskscan -v';
+    assert_script_run 'lvscan -v';
+    assert_script_run 'pvs -a';
+
+    my @volumes = split(/\n/, script_output q[lvscan | awk '{print $2}'| sed s/\'//g]);
+    
+    foreach my $volume (@volumes) {
+        chomp;
+        my $lvdisp_output = script_output "lvdisplay $volume";
+        foreach (keys %{$expected_lv_stats}) {
+            die "Value of $expected_lv_stats->{$_} was not found in $volume configuration" unless ($lvdisp_output =~ /(?<tested_string>$expected_lv_stats->{$_})/);
+            record_info('TEST', "Found $+{tested_string} in $volume");
+        }
+    }
+
+    record_info('INFO', 'Create a file on home volume');
+    my $test_file = '/home/bernhard/test_file.txt';
+    assert_script_run 'df -h  | tee original_usage';
+    assert_script_run "dd if=/dev/zero of=$test_file count=1024 bs=1M";
+    assert_script_run "ls -lah $test_file";
+    sleep 25;
+    assert_script_run 'df -h | tee instant_usage';
+    
+    if ((script_run 'diff original_usage instant_usage') != 1) {
+        die "Disk usage do not differ!";
+    }
+      
+    record_info('INFO', 'Check partitions');
+    my @expected_partitions = (
+        "/",
+        "/.snapshots",
+	"/tmp",
+	"/var",
+        "/opt",
+	"/home",
+	"/usr\/local",
+        "/srv",
+	"/root"
+    );
+
+    my @partitions = split(/\n/, script_output q[df -h | grep mapper | awk '{print $6}']);
+
+    foreach my $partition (@partitions) {
+        chomp;
+        die "$partition was not found in filesystem" unless (grep {$partition eq $_ } @expected_partitions);
```

Comment:
I would suggest doing it differently, so that test reports all missing partitions instead of failing once first is identified.

Reply:
Can you suggest something? how do you tackle things like that in perl?

Reply:
Something like:
```perl
my $errors = '';
foreach my $partition (@partitions) {
$errors .= "$partition was not found in filesystem" unless (grep {$partition eq $_ } @expected_partitions);
}
die "Errors identified during partitions check:\n $errors" if $errors;
```